core/src/main/resources/lib/hudson/buildProgressBar.jelly

```diff
@@ -26,6 +26,7 @@ THE SOFTWARE.
   <st:documentation>
     Progress bar for a build in progress.
 
+    <!-- TODO: Do Functions.getConsoleUrl and ConsoleUrlProvider need to be adapted to use Queue.Executable instead of Run? -->
```

Comment:
I think making that change would be fine, but I will investigate this case to understand if it is necessary.

Reply:
I did some quick testing with `node` steps and Pipeline and `PlaceholderExecutable` does seem to hit this case. I think we can probably special-case this particular call (either here or in `Functions.getConsoleUrl`) rather than needing to change the API itself though.

Reply:
Well after some very annoying debugging I finally realized that the `node` step defines its own `executorCell.jelly`, which actually misuses this taglib and passes a `Queue.Task` (`PlaceholderTask`) instead of `Queue.Executable` (`PlaceholderExecutable`), see https://github.com/jenkinsci/workflow-durable-task-step-plugin/blob/94bdc2b50c6fbab37526804d232eb2f1da971db9/src/main/resources/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution/PlaceholderTask/PlaceholderExecutable/executorCell.jelly#L35. So, to fix this here in core as-is, we'd have to make `Functions.getConsoleUrl` take `Object`, and then extract the `Run` from either a `Queue.Task` or `Queue.Executable`.

Reply:
I think I'll file a PR against `workflow-durable-task-step` though and just handle `Queue.Executable` here as documented.

Reply:
See https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/330.

Reply:
Addressed in [46b9cda](https://github.com/jenkinsci/jenkins/pull/8321/commits/46b9cda076f1184c846dc4c816c8568982719de0).